pkg/install/4-addcustomisations.go

```diff
@@ -0,0 +1,95 @@
+package install
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	operatorclient "github.com/openshift/client-go/operator/clientset/versioned"
+	consoleapi "github.com/openshift/console-operator/pkg/api"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/Azure/ARO-RP/pkg/util/restconfig"
+)
+
+func (i *Installer) addCustomisations(ctx context.Context) error {
+	restConfig, err := restconfig.RestConfig(ctx, i.env, i.doc.OpenShiftCluster)
+	if err != nil {
+		return err
+	}
+
+	{
+		cli, err := configclient.NewForConfig(restConfig)
+		if err != nil {
+			return err
+		}
+
+		i.log.Print("waiting for version clusterversion")
+		timeoutCtx, cancel := context.WithTimeout(ctx, 30*time.Minute)
+		defer cancel()
+		err = wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
+			cv, err := cli.ConfigV1().ClusterVersions().Get("version", metav1.GetOptions{})
+			if err == nil {
+				for _, cond := range cv.Status.Conditions {
+					if cond.Type == configv1.OperatorAvailable && cond.Status == configv1.ConditionTrue {
+						return true, nil
+					}
+				}
+			}
+			return false, nil
+
+		}, timeoutCtx.Done())
+		if err != nil {
+			return err
+		}
+
+		i.log.Print("disabling updates")
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			cv, err := cli.ConfigV1().ClusterVersions().Get("version", metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+
+			cv.Spec.Upstream = ""
+			cv.Spec.Channel = ""
+
+			_, err = cli.ConfigV1().ClusterVersions().Update(cv)
+			return err
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	{
+		i.log.Print("updating console branding")
+		cli, err := operatorclient.NewForConfig(restConfig)
+		if err != nil {
+			return err
+		}
+
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			operatorConfig, err := cli.OperatorV1().Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+
+			operatorConfig.Spec.Customization.Brand = operatorsv1.BrandAzure
+
+			_, err = cli.OperatorV1().Consoles().Update(operatorConfig)
+			return err
+		})
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
is there something here that we should wait for?  e.g. can we wait for an indication that the console has reloaded with the new branding before calling the cluster creation complete?

Reply:
@jim-minter good question! Waiting for the console to reaload with ARO branding will add to the cluster creation time. 

Given that console should be accessible during the reload, is it critical if user can see the standard branding for a minute or two?

Reply:
please start by adding in the wait - let's see where we go from there.